src/cc65/declare.c

```diff
@@ -681,15 +681,17 @@ static int ParseFieldWidth (Declaration* Decl)
         return -1;
     }
 
-    /* Read the width */
-    NextToken ();
-    ConstAbsIntExpr (hie1, &Expr);
     if (!IsClassInt (Decl->Type)) {
         /* Only integer types may be used for bit-fields */
-        Error ("Bit-field has invalid type '%s'", GetBasicTypeName (Decl->Type));
+        Error ("Bit-field has invalid type '%s', must be integral",
+               GetBasicTypeName (Decl->Type));
         return -1;
     }
 
+    /* Read the width */
+    NextToken ();
+    ConstAbsIntExpr (hie1, &Expr);
```

Comment:
Since you have been changing this part... Could the width-parsing part be moved closer to the check that uses it (`Expr.IVal`), after the type size check (`SizeOf (Decl->Type)`) on Line 695?

Reply:
Done.